urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/jasl/utilities/Serialization.java

```diff
@@ -0,0 +1,140 @@
+// ************************************************************************** //
+// Serialization.java - This class is part of the <B>utilities</B> package,          //
+//                      which contains the definitions of objects used to     //
+//                      support the classes more directly associated with the //
+//                      game itself.                                          //
+//                                                                            //
+// Written By: Craig R. Campbell  -  March 2013                               //
+// ************************************************************************** //
+
+package jasl.utilities;
+
+import java.io.*; // For [File|Object][In|Out]putStream classes.
+
+/**
+ * This class provides static methods used to serialize a specified object into
+ * a file, as well as to read a file and deserialize its contents into an
+ * object.
+ *
+ * @version 1.0
+ * @author Copyright (C) 2013 Craig R. Campbell (craigonic@gmail.com)
+ * @see <A HREF="../../../source/jasl/utilities/Serialization.html">Source code</A>
+ */
+
+public final class Serialization
+{
+	// Symbolic constants
+
+	private static final String CLASS_NAME = "Serialization";
+
+	// Public static methods
+
+	/**
+	 * Write the specified object to the indicated file.
+	 *
+	 * @param object the object to be serialized.
+	 * @param filename the name of the file that the serialized data
+	 * should be written to.
+	 *
+	 * @throws NullPointerException in the case of a null object or file name
+	 * @throws IllegalArgumentException in the case of a zero length file
+	 * name
+	 */
+
+	public static final void serializeToFile(Object object,String filename)
+	{
+		// Define local constants.
+
+		String METHOD_NAME = "serializeToFile";
+
+		// Check the parameters received and throw the appropriate
+		// exception if necessary.
+
+		if ((null == object) || (null == filename))
+		{
+			throw new NullPointerException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                          METHOD_NAME,
+			                                                          Messages.NULL_PARAMETER_MSG));
+		}
+
+		if (0 == filename.length())
+		{
+			throw new IllegalArgumentException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                              METHOD_NAME,
+			                                                              Messages.ZERO_LENGTH_PARAMETER_MSG));
+		}
+
+		try
+		{
+			ObjectOutputStream objectOutputStream =
+				new ObjectOutputStream(new FileOutputStream(filename));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource created by the following constructor: *FileOutputStream*. The resource is referenced at lines 70, 71. The resource is closed at line 72. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure the object returned by *FileOutputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).